docs/workflow/testing/libraries/testing.md

```diff
@@ -101,4 +101,21 @@ dotnet build /t:Test /p:Outerloop=true
 
 ### Running tests on a different target framework
 
-Each test project can potentially have multiple target frameworks. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `TargetFrameworks` property specifies the valid target frameworks.
\ No newline at end of file
+Each test project can potentially have multiple target frameworks. There are some tests that might be OS-specific, or might be testing an API that is available only on some target frameworks, so the `TargetFrameworks` property specifies the valid target frameworks.
+
+### Running tests in custom compilation modes
```

Comment:
@ViktorHofer I'm interested in any particular details I missed here. We really need some documentation on this sort of thing, as the behaviors of our tests are not particularly clear to folks who don't look into these systems often.